eclipse-theia/theia
# eslint: update rules for 'theia.d.ts'

#### What it does


The following commit enables linting of the 'theia.d.ts' file which had many linting violation and caused issues during reviews. The code has also been fixed to meet linting rules, and should be fine for the future.

#### How to test


Verify that the build successfully passes, linting should be fixed for `theia.d.ts`.

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: vince-fugnitto <vincent.fugnitto@ericsson.com>
